game_board.gd

```diff
@@ -204,22 +209,25 @@ func _select_active_unit(cell: Vector2) -> void:
 	# registered in the `cell`.
 	if not _units.has(cell):
 		return
-
-	if _units[cell].isPlayerControllable:
+#todo: fix previously freed error on next line when clicking on tile after the occupant dies
+	if is_occupied(cell) && _units[cell].isPlayerControllable && _units[cell].actionPoints > 0:
```

Comment:
this is where the dead unit bug I mentioned in discord is, specifically on isPlayerControllable, so is_occupied probably needs to be tweaked